torch/testing/_internal/distributed/rpc/rpc_test.py

```diff
@@ -4869,10 +5057,139 @@ def test_device_maps_remote(self):
             args=(torch.zeros(2), 1)
         )
 
+        self.assertEqual(rref.to_here().device.index, 1)
         self.assertEqual(rref.to_here(), torch.ones(2).to(1))
 
         rpc.shutdown()
 
+    @staticmethod
+    def _slow_add_on_user_stream(x, y):
+        s0 = torch.cuda.current_stream(x.device)
+        s1 = torch.cuda.Stream(device=x.device)
+        with torch.cuda.stream(s1):
+            torch.cuda._sleep(10 * FIFTY_MIL_CYCLES)
+            z = x + y
```

Comment:
Shouldn't there also be a synchronization before the addition? `x` and `y` might still be filled in, and this is being done in the current streams, hence it's only safe to access them from the current stream, or from streams that are explicitly synchronized with the current one.

Also, we should check that `x` and `y` are on the same devices, or else we need to also sync with the current stream of `y.device`.

Reply:
ah, yes, good catch! it probably was hided by the `_sleep`